From david-b@pacbell.net Fri Dec  8 12:50:21 2006
From: Wojtek Kaniewski <wojtekka@toxygen.net>
Subject: USB: at91_udc: allow drivers that support high speed
Date: Fri, 8 Dec 2006 09:39:36 -0800
To: Greg KH <greg@kroah.com>
Cc: Wojtek Kaniewski <wojtekka@toxygen.net>
Message-Id: <200612080939.37486.david-b@pacbell.net>


This patch allows gadget drivers that support high speed (e.g. gadgetfs)
to work properly with at91_udc.

Fix suggested by Milan Svoboda in
http://marc.theaimsgroup.com/?l=linux-usb-devel&m=115822184711817

Signed-off-by: Wojtek Kaniewski <wojtekka@toxygen.net>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Cc: Andrew Victor <andrew@sanpeople.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/gadget/at91_udc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/at91_udc.c
+++ gregkh-2.6/drivers/usb/gadget/at91_udc.c
@@ -1579,7 +1579,7 @@ int usb_gadget_register_driver (struct u
 	int		retval;
 
 	if (!driver
-			|| driver->speed != USB_SPEED_FULL
+			|| driver->speed < USB_SPEED_FULL
 			|| !driver->bind
 			|| !driver->setup) {
 		DBG("bad parameter.\n");
